go/libraries/doltcore/sqle/database.go

```diff
@@ -48,47 +51,54 @@ func (db *Database) Name() string {
 	return db.name
 }
 
-// Tables returns the tables in this database, currently exactly the same tables as in the current working root.
-func (db *Database) Tables() map[string]sql.Table {
-	ctx := context.Background()
+func (db *Database) GetTableInsensitive(ctx context.Context, tblName string) (sql.Table, bool, error) {
+	lwrName := strings.ToLower(tblName)
+	if strings.HasPrefix(lwrName, DoltDiffTablePrefix) {
+		tblName = tblName[len(DoltDiffTablePrefix):]
+		dt, err := NewDiffTable(tblName, db.dEnv)
+
+		if err != nil {
+			return nil, false, err
+		}
+
+		return dt, true, nil
+	}
+
+	if lwrName == LogTableName {
+		return NewLogTable(db.dEnv), true, nil
+	}
 
-	tables := make(map[string]sql.Table)
 	tableNames, err := db.root.GetTableNames(ctx)
 
-	// TODO: fix panics
 	if err != nil {
-		panic(err)
+		return nil, false, err
 	}
 
-	for _, name := range tableNames {
-		table, ok, err := db.root.GetTable(ctx, name)
+	exactName, ok := sql.GetTableNameInsensitive(tblName, tableNames)
 
-		// TODO: fix panics
-		if err != nil {
-			panic(err)
-		}
+	if !ok {
+		return nil, false, nil
+	}
 
-		if !ok {
-			panic("Error loading table " + name)
-		}
+	tbl, ok, err := db.root.GetTable(ctx, exactName)
 
-		sch, err := table.GetSchema(ctx)
-		// TODO: fix panics
-		if err != nil {
-			panic(err)
-		}
+	if err != nil {
+		return nil, false, err
+	} else if !ok {
+		panic("Name had already been verified... This is a bug")
```

Comment:
Probably want to include the string being matched in the panic message

Reply:
updated